optuna/integration/_lightgbm_tuner/optimize.py

```diff
@@ -565,12 +568,31 @@ def tune_feature_fraction(self, n_trials: int = 7) -> None:
         self._tune_params([param_name], len(param_values), sampler, "feature_fraction")
 
     def tune_num_leaves(self, n_trials: int = 20) -> None:
-        self._tune_params(["num_leaves"], n_trials, optuna.samplers.TPESampler(), "num_leaves")
+        if self._optuna_seed is None:
+            self._tune_params(["num_leaves"], n_trials, optuna.samplers.TPESampler(), "num_leaves")
+        else:
+            self._tune_params(
+                ["num_leaves"],
+                n_trials,
+                optuna.samplers.TPESampler(seed=self._optuna_seed),
+                "num_leaves",
+            )
```

Comment:
Because the default value of the `seed` argument of `TPESampler` is `None`, I think we can merge these lines as follows. Do you have any reasons to split them?
```suggestion
        self._tune_params(["num_leaves"], n_trials, optuna.samplers.TPESampler(seed=self._optuna_seed), "num_leaves")
```

Reply:
Thank you for applying my suggestion.